src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -179,7 +179,10 @@ private function getCurlCommand(array $trace): ?string
         }
 
         $debug = explode("\n", $trace['info']['debug']);
-        $url = self::mergeQueryString($trace['url'], $trace['options']['query'] ?? [], true);
+        $url = implode('', self::resolveUrl(
+            self::parseUrl($trace['url'], $trace['options']['query'] ?? []),
+            isset($trace['options']['base_uri']) ? self::parseUrl($trace['options']['base_uri']) : null
```

Comment:
I think we should parse $debug instead because options won't always contain the base_uri, when default options are used.

Reply:
@nicolas-grekas Done